easybuild/easyblocks/c/cppcheck.py

```diff
@@ -0,0 +1,139 @@
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+@author: Damian Alvarez (Forschungszentrum Juelich GmbH)
+"""
+import os
+import shutil
+
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.framework.easyconfig import BUILD
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.run import run_cmd
+
+
+class EB_cppcheck(ConfigureMake):
+    """
+    EasyBlock to install cppcheck
+    """
+    
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """
+        Define if we are using rules or not, and if we are building the GUI
+        """
+        extra = {
+            'have_rules': [False, "Use rules", BUILD],
+            'build_gui': [False, "Build GUI", BUILD],
+        }
+        if extra_vars is None:
+            extra_vars = {}
+        extra.update(extra_vars)
+        return ConfigureMake.extra_options(extra_vars=extra)
+
+
+    def configure_step(self):
+        """
+        Run qmake on the GUI, if necessary
+        """
+        if self.cfg['build_gui']:
+            os.chdir(self.cfg['start_dir']+'/gui')
+            if self.cfg['have_rules']:
+                cmd = 'qmake HAVE_RULES=yes'
+            else:
+                cmd = 'qmake'
+            run_cmd(cmd, log_all=True, simple=True)
+            os.chdir(self.cfg['start_dir'])
+        else:
+            pass
+
+    def build_step(self, verbose=False):
+        """
+        Compile cppcheck with make and cppcheck-gui with qmake and make 
+        """
+        self.cfg.update('buildopts', 'CFGDIR=%(installdir)s/cfg')
+        
+        if self.cfg['have_rules']:
+            self.cfg.update('buildopts', 'HAVE_RULES=yes')
+
+        super(EB_cppcheck, self).build_step(verbose=verbose)
+        
+        if self.cfg['build_gui']:
+            super(EB_cppcheck, self).build_step(verbose=verbose,path='gui')
+
+    def install_step(self):
+        """
+        Install cppcheck with make install and cppcheck-gui copying the file
+        """
+        self.cfg.update('installopts', 'DESTDIR=%(installdir)s/ PREFIX="" CFGDIR=cfg')
+        
+        super(EB_cppcheck, self).install_step()
+       
+        # The GUI doesn't have make install, so we copy it manually
+        if self.cfg['build_gui']:
+            try:
+                # make sure we're (still) in the start dir
+                os.chdir(self.cfg['start_dir'])
+
+                file_to_copy = 'cppcheck-gui'
+                self.log.debug("Copying cppcheck-gui")
+
+                filepath = self.cfg['start_dir'] + '/gui/' + file_to_copy
```

Comment:
please use `os.path.join(self.cfg['start_dir'], 'gui', 'cpp-check-gui')` (there's no need for `file_to_copy` since it's used only once?)